### ISSUE TITLE
Extend the gateway protocol to create a process instance with start instructions

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#9396]
We need to extend the gateway protocol to create a process instance with start instructions.

Specifically, we need to extend the [CreateProcessInstanceRequest](https://github.com/camunda/zeebe/blob/8.0.2/gateway-protocol/src/main/proto/gateway.proto#L87-L101) with the start instructions from #9388.

Make sure to document how we can extend the gateway protocol.

Blocked by: #9388

## Out of Scope
- response containing created element instance keys (not necessary for a first iteration)
